cpp/src/io/parquet/reader_impl.cpp

```diff
@@ -607,21 +611,26 @@ table_with_metadata reader::impl::read_chunk()
     }
   }
 
-  prepare_data(0 /*skip_rows*/,
-               std::nullopt /*num_rows, `nullopt` means unlimited*/,
-               true /*uses_custom_row_bounds*/,
-               {} /*row_group_indices, empty means read all row groups*/,
-               std::nullopt /*filter*/);
-  return read_chunk_internal(true, std::nullopt);
+  table_metadata metadata;
+  populate_metadata(metadata);
+  auto expr_conv     = named_to_reference_converter(filter, metadata);
+  auto output_filter = expr_conv.get_converted_expr();
+
+  prepare_data(skip_rows, num_rows, true, row_group_indices, output_filter);
```

Comment:
Should `uses_custom_row_bounds` argument be hardcoded to true in case of chunked reader or should it come from `options`?

Reply:
Answered by @ttnghia 

The original answer came from Dave Baranec when I put in the hard coded true value:

> There's a couple other bugs.  We are calling read_chunk_internal() in a few places where we're only passing uses_custom_row_bounds.  But if chunked_read_size is > 0 we need to force that to be true.   If we don't, various important things don't happen for all the remaining chunks.
Basically if this code doesn't execute, it'll be bad :slightly_smiling_face:

https://github.com/rapidsai/cudf/blob/bd7b5101cfb77b5b0210cc1d7b78a587821b9bf6/cpp/src/io/parquet/reader_preprocess.cu#L499 

So it is indeed a bit confusing. We indeed should not hard code it as true  as it is not correct, but if we change that then we need to fix all the related stuffs.

Reply:
@vuule I think I am going to hardcode to `true` in chunked reader for now (24.06) and add these comments there as `TODO` + open an issue for 24.08. Does that sound ok?